test/testing_util.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"os"
+	"os/exec"
+
+	"fmt"
+
+	"k8s.io/kops/util/pkg/vfs"
+)
+
+// KopsTest
+type KopsTest struct {
+	ClusterName  string
+	StateStore   string
+	S3BucketName string
+
+	Image      string
+	DomainName string
+	NodeUpURL  string
+
+	NodeCount   int
+	MasterCount int
+	NodeZones   string
+	MasterZones string
+	NodeSize    string
+	MasterSize  string
+	Verbosity   int
+	Networking  string
+	Topology    string
+
+	K8sVersion  string
+}
+
+// Basic Pre Test
+func (t *KopsTest) Pre() (*KopsTest, error) {
+	t, err := t.basicPreCheck()
+	if err != nil {
+		return nil, fmt.Errorf("error in precheck %v", err)
+	}
+	t, err = t.createBucket()
+	if err != nil {
+		return nil, fmt.Errorf("error in create bucket %v", err)
+	}
+	t, err = t.createClusterName()
+	if err != nil {
+		return nil, fmt.Errorf("error in create cluster name %v", err)
+	}
+
+	// TODO: setup dynamically
```

Comment:
RE: Dynamically generated.. it would be *fabulous* if our tests could read in a cluster spec YAML file...